### PR TITLE
feat: update BackTop

### PR DESCRIPTION
**Pull request by liulinboyi**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/back-top/index.jsx [hunk 1]
```diff
@@ -64,7 +58,7 @@ const BackTop = {
       scrollTo(0, {
         getContainer: target,
       });
-      this.$emit('click', e);
+      this.$emit('top', e);
```
Comment (liulinboyi) [inline components/back-top/index.jsx:61]: 自定义事件名称如果为"click"，点击按钮，回调函数会执行两次。

**Comment by liulinboyi** [comment]
@tangjinzhou  Please help review, thanks!

File: components/back-top/index.jsx [hunk 2]
```diff
@@ -104,9 +99,8 @@ const BackTop = {
 };
```
Comment (tangjinzhou) [inline components/back-top/index.jsx:99]: $slots.default
to
$slots.default && $slots.default()

File: components/back-top/index.jsx [hunk 3]
```diff
@@ -89,7 +85,6 @@ const BackTop = {
     );
     const divProps = {
       on: {
-        ...getListeners(this),
         click: this.scrollToTop,
```
Comment (tangjinzhou) [inline components/back-top/index.jsx:88]: 扁平化处理 
divProps = {
 onClick: this.scrollToTop
...
}

File: components/back-top/index.jsx (hunk 1 above)
Comment (tangjinzhou) [reply components/back-top/index.jsx:63]: 方案1: 设置 inheritAttrs: false，并单独处理 class style等
方案2: 属性声明中加入 onClick
改成 方案2 方式吧

**Comment by liulinboyi** [comment]
@tangjinzhou ok all resolved.

**Pull request merged**